src/Serializer/CollectionNormalizer.php

```diff
@@ -0,0 +1,218 @@
+
+        $this->identityMap->reset($collection['id']);
+
+        if (empty($this->globalCallback)) {
+            $this->globalCallback = new CallbackPromise(function () {
+                $this->identityMap->fillMissingWith(function ($id) {
+                    return $this->collectionsClient->getCollection(
+                        ['Accept' => new MediaType(CollectionsClient::TYPE_COLLECTION, 1)],
+                        $id
+                    );
+                });
+
+                $this->globalCallback = null;
+
+                return $this->identityMap->waitForAll();
+            });
+        }
+
+        return $this->globalCallback
+            ->then(function (array $collections) use ($collection) {
+                return $collections[$collection['id']];
+            });
+    }
+
+    public function normalize($object, $format = null, array $context = []) : array
+    {
+        $normalizationHelper = new NormalizationHelper($this->normalizer, $this->denormalizer, $format);
+
+        $data = [];
+        $data['id'] = $object->getId();
+        $data['title'] = $object->getTitle();
+        if ($object->getSubTitle()) {
+            $data['subTitle'] = $object->getSubTitle();
+        }
+        if ($object->getImpactStatement()) {
+            $data['impactStatement'] = $object->getImpactStatement();
+        }
+        $data['updated'] = $object->getPublishedDate()->format(DATE_ATOM);
+
+        $data['image']['thumbnail'] = $this->normalizer->normalize($object->getThumbnail(), $format, $context);
+        if (count($object->getSubjects()) > 0) {
+            $data['subjects'] = $normalizationHelper->normalizeSequenceToSnippets($object->getSubjects(), $context);
+        }
+
+        $data['selectedCurator'] = $normalizationHelper->normalizeToSnippet($object->getSelectedCurator());
+        if ($object->selectedCuratorEtAl()) {
+            $data['selectedCurator']['etAl'] = $object->selectedCuratorEtAl();
+        }
+
+        if (empty($context['snippet'])) {
+            $data['image']['banner'] = $this->normalizer->normalize($object->getBanner(), $format, $context);
+
+            $data['curators'] = $normalizationHelper->normalizeSequenceToSnippets($object->getCurators(), $context);
+
+            $contentNormalization = function ($eachContent) use ($normalizationHelper) {
+                $eachContentData = $normalizationHelper->normalizeToSnippet($eachContent);
+                $contentClasses = [
+                    ArticlePoA::class => 'research-article',
+                    ArticleVoR::class => 'research-article',
+                    BlogArticle::class => 'blog-article',
+                    Interview::class => 'interview',
+                ];
+                if (!array_key_exists(get_class($eachContent), $contentClasses)) {
+                    throw new LogicException('Class of content '.get_class($eachContent).' is not supported in a Collection. Supported classes are: '.var_export($contentClasses, true));
```

Comment:
will throw in case a new content type is added to the collection `content` or `relatedContent`, in that case we won't know what its `type` value should be